Added support for Boolean and Uuid data types to RowReader/RowWriter

cmhodgson opened the pull request.

cmhodgson committed:
Added support for Boolean and Uuid data types to RowReader/RowWriter
Other minor improvements including autocloseable support and error handling
Incremented version to 1.2.0

cmhodgson committed:
commit of v1.3.0 to support new java data processing tools in ols-geocoder-process

bolyachevets committed:
Merge branch 'main' into dev

bolyachevets commented on src/main/java/ca/bc/gov/ols/rowreader/AbstractBasicWktRowReader.java:
```diff
@@ -0,0 +1,149 @@
+import org.locationtech.jts.io.ParseException;
+import org.locationtech.jts.io.WKTReader;
+
+public abstract class AbstractBasicWktRowReader extends AbstractBasicRowReader {
```
Correct me if I am wrong, but looks like some methods this class is overriding (e.g getInt, getBoolean, getDate, etc.) are not any different from the ones in AbstractBasicRowReader. Can we just drop corresponding implementations here?

cmhodgson replied:
You are entirely correct. There were also a few other things in there that I cleaned up as well.

cmhodgson committed:
- removed duplicated reader code
- moved x/y-column-based point handling code into XsvReader where it should really be
- made use of defaultGeometryColumn as intended
